testing/fleetservertest/fleetserver_test.go

```diff
@@ -40,28 +146,523 @@ func ExampleNewServer_status() {
 
 func ExampleNewServer_checkin() {
 	agentID := "agentID"
+	tmpl := TmplPolicy{ActionID: "anActionID"}
+
+	actions, err := NewActionPolicyChangeWithFakeComponent(tmpl)
+	if err != nil {
+		panic(fmt.Sprintf("failed to get new actions: %v", err))
+	}
+
+	// NewHandlerCheckinFakeComponent
+	ts := NewServer(&Handlers{
+		CheckinFn: NewHandlerCheckinFakeComponent(func() (CheckinAction, *HTTPError) {
+			return CheckinAction{Actions: []string{actions}}, nil
+		})},
+		WithAgentID(agentID))
+
+	cmd := fleetapi.NewCheckinCmd(
+		agentInfo(agentID), sender{url: ts.URL, path: NewPathCheckin(agentID)})
+
+	got, _, err := cmd.Execute(context.Background(), &fleetapi.CheckinRequest{})
+	if err != nil {
+		panic(fmt.Sprintf("ExampleNewServer_checkin failed executing checkin: %v", err))
+	}
+
+	fmt.Println(got.Actions)
+	// Output:
+	// [action_id: anActionID, type: POLICY_CHANGE]
+}
+
+func ExampleNewServer_ack() {
+	agentID := "agentID"
+	actionID := "actionID"
 
-	ts := NewServer(API{
-		CheckinFn: NewCheckinHandler(agentID, "", false),
+	handlers := &Handlers{
+		AckFn: NewHandlerAck(),
+	}
+	ts := NewServer(handlers)
+
+	cmdAck := fleetapi.NewAckCmd(
+		agentInfo(agentID), sender{url: ts.URL, path: NewPathAgentAcks(agentID)})
+
+	respAck, err := cmdAck.Execute(context.Background(),
+		&fleetapi.AckRequest{Events: []fleetapi.AckEvent{
+			{
+				EventType: "ACTION_RESULT",
+				SubType:   "ACKNOWLEDGED",
+				Timestamp: "2022-12-01T01:02:03.00004-07:00",
+				ActionID:  actionID,
+				AgentID:   agentID,
+				Message: fmt.Sprintf("Action '%s' of type 'ACTION_TYPE' acknowledged.",
+					actionID),
+			},
+			{
+				EventType: "ACTION_RESULT",
+				SubType:   "ACKNOWLEDGED",
+				Timestamp: "2022-12-01T01:02:03.00004-07:00",
+				ActionID:  "not-received-on-checkin",
+				AgentID:   agentID,
+				Message: fmt.Sprintf("Action '%s' of type 'unknwon_action' acknowledged.",
+					"not-received-on-checkin"),
+			}}})
+	if err != nil {
+		panic(fmt.Sprintf("failed executing checkin: %v", err))
+	}
+	fmt.Printf("%#v\n", respAck)
+
+	// Output:
+	// &fleetapi.AckResponse{Action:"acks", Errors:false, Items:[]fleetapi.AckResponseItem{fleetapi.AckResponseItem{Status:200, Message:"OK"}, fleetapi.AckResponseItem{Status:200, Message:"OK"}}}
+}
+
+func ExampleNewServer_enrol() {
```

Comment:
For consistency with the other suggestions

```suggestion
func ExampleNewServer_enroll() {
```